idaes/generic_models/properties/core/coolprop/coolprop_forms.py

```diff
@@ -0,0 +1,226 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+This module contains methods for constructing CoolProp expressions.
+"""
+from pyomo.environ import exp, units as pyunits, Var
+from pyomo.core.expr.calculus.derivatives import Modes, differentiate
+
+from idaes.core.util.exceptions import ConfigurationError
+
+
+# TODO : Only have temperature derivative expression for exponential_tau form
+# TODO : Add other derivative forms as/if required
+
+def parameters_nt_sum(cobj, prop, nlist, tlist):
+    """
+    Method for creating parameters for expression forms using n-t parameters
+
+    Args:
+        cobj - component object that will contain the parameters
+        prop - name of property parameters are associated with
+        nlist - list of values for n-parameter
+        tlist - list of values for t-parameter
+
+    Returns:
+        None
+    """
+    if len(nlist) != len(tlist):
+        raise ConfigurationError(
+            f"{cobj.name} mismatched length between n and t parameters "
+            f"for CoolProp exponential form for property {prop}. Please "
+            f"ensure the number of n and t parameters are equal.")
+
+    for i, nval in enumerate(nlist):
+        coeff = Var(doc="Multiplying parameter for CoolProp exponential form",
+                    units=pyunits.dimensionless)
+        cobj.add_component(prop+"_coeff_n"+str(i+1), coeff)
+        coeff.fix(nval)
+
+    for i, tval in enumerate(tlist):
+        coeff = Var(doc="Exponent parameter for CoolProp exponential form",
+                    units=pyunits.dimensionless)
+        cobj.add_component(prop+"_coeff_t"+str(i+1), coeff)
+        coeff.fix(tval)
+
+
+def _nt_sum(cobj, prop, theta):
+    """
+    Method for creating sum expressions in n-t forms (sum(n[i]*theta**t[i]))
+
+    Args:
+        cobj - component object that will contain the parameters
+        prop - name of property parameters are associated with
+        theta - expression or variable to use for theta in expression
+
+    Returns:
+        Pyomo expression of sum term
+    """
+    # Build sum term
+    i = 1
+    s = 0
+    while True:
+        try:
+            ni = getattr(cobj, prop+"_coeff_n"+str(i))
```

Comment:
nitpick: python 3 f-string would be `f"{prop}_coeff_n{i}"` which I think is cleaner